app/uk/gov/hmrc/gform/formtemplate/FormTemplateService.scala

```diff
@@ -58,8 +58,8 @@ class FormTemplateService(formTemplateRepo: Repo[FormTemplate], formTemplateRawR
 
   def verifyAndSave(formTemplate: FormTemplate): FOpt[Unit] =
     for {
-      _   <- verify(formTemplate)
-      _   <- formTemplateRepo.upsert(mkSpecimen(formTemplate))
-      res <- formTemplateRepo.upsert(formTemplate)
+      formTemplateToSave <- verify(formTemplate)
+      _                  <- formTemplateRepo.upsert(mkSpecimen(formTemplateToSave))
+      res                <- formTemplateRepo.upsert(formTemplateToSave)
```

Comment:
what is the benefit of this change?

Reply:
@girishkamat Is this more clear? It is still not perfect separation, since `rewrite` still does a validation + rewriting. It is so because validation is not doing any type refinement so rewriting would need to do exactly same verification as validation would do.